### PR TITLE
Port Sugar ORM to Room

### PR DESCRIPTION
**Pull request by jatinluthra14**
This PR will have some conflicts as this feature is merged already.
It is done for GCI and learning purposes and is not for merging

### CODE REVIEW

**Comment by m-murad** [comment]
Please resolve the merge conflicts.

**Comment by m-murad** [comment]
Please resolve conflicts.

**Comment by jatinluthra14** [comment]
@m-murad Hi, this feature is merged already, It is not to be merged. Just for learning and GCI. So there would be conflicts. Please ignore them.

File: app/src/main/java/com/community/jboss/leadmanagement/AppDatabase.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                Number.class,
+                Tag.class
+        },
+        version = 1,
+        exportSchema = false)
+public abstract class AppDatabase extends RoomDatabase {
+    public abstract ContactDAO contactDAO();
+    public abstract NumberDAO numberDAO();
+    public abstract TagDAO tagDAO();
+}
```
Comment (shubhamvashisht) [inline app/src/main/java/com/community/jboss/leadmanagement/AppDatabase.java]: add a new blank line.

File: app/src/main/java/com/community/jboss/leadmanagement/Classes/NumberDAO.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @Query("SELECT * FROM Number WHERE number LIKE :number")
+    List<Number> getNumberusingNumber(String number);
+
+    @Query("SELECT * FROM Number WHERE number LIKE :contact")
+    List<Number> getNumberusingContact(String contact);
+
+    @Update
+    void updateNumber(Number... numbers);
+
+}
```
Comment (shubhamvashisht) [inline app/src/main/java/com/community/jboss/leadmanagement/Classes/NumberDAO.java]: Add a blank line.

**Comment by jatinluthra14** [comment]
@shubhamvashisht Done and squashed

File: app/src/main/java/com/community/jboss/leadmanagement/AppDatabase.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Created by jatin on 26-12-2017.
+ */
+
+@Database(entities =
+        {
+                Contact.class,
+                Number.class,
+                Tag.class
+        },
```
Comment (aashutoshrathi) [inline app/src/main/java/com/community/jboss/leadmanagement/AppDatabase.java]: Use tab of length 4.

**Comment by jatinluthra14** [comment]
@aashutoshrathi Done and squashed.

**Pull request closed without merging**